src/bank.rs

```diff
@@ -1490,4 +1544,17 @@ mod tests {
             Ok(_)
         );
     }
+    #[test]
+    fn test_last_id_age() {
```

Comment:
A better testing style would be to break this up into three cases, an empty queue that tests the not-found case, a singleton that tells us that the most recent ID is age zero, and the induction step that tells us order matters and that it's a queue, not a stack.  This test only checks that third case. For the second case, the age appears to be 1, not 0, why?